bigquery/src/main/java/org/apache/zeppelin/bigquery/BigQueryRestrictedAccessInterpreter.java

```diff
@@ -0,0 +1,299 @@
+import org.apache.zeppelin.interpreter.InterpreterResult;
+import org.apache.zeppelin.interpreter.thrift.InterpreterCompletion;
+import org.apache.zeppelin.scheduler.Scheduler;
+import org.apache.zeppelin.scheduler.SchedulerFactory;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Properties;
+
+
+/**
+ * BigQuery interpreter for Zeppelin.
+ * <p>
+ * <ul>
+ * <li>{@code zeppelin.bigquery.project_id} - Project ID in GCP</li>
+ * <li>{@code zeppelin.bigquery.wait_time} - Query Timeout in ms</li>
+ * <li>{@code zeppelin.bigquery.max_no_of_rows} - Max Result size</li>
+ * </ul>
+ * <p>
+ * <p>
+ * How to use: <br/>
+ * {@code %bigquery.sql<br/>
+ * {@code
+ * SELECT departure_airport,count(case when departure_delay>0 then 1 else 0 end) as no_of_delays
+ * FROM [bigquery-samples:airline_ontime_data.flights]
+ * group by departure_airport
+ * order by 2 desc
+ * limit 10
+ * }
+ * </p>
+ */
+
+public class BigQueryRestrictedAccessInterpreter extends Interpreter {
+  private static final String LEGACY_SQL = "#legacySQL";
+  private static final int HTTP_TIMEOUT = 10000;
+  private static Logger logger = LoggerFactory.getLogger(BigQueryInterpreter.class);
+  private static BigQuery service = null;
+  //Mutex created to create the singleton in thread-safe fashion.
+  private static Object serviceLock = new Object();
+
+  static final String PROJECT_ID = "zeppelin.bigquery.project_id";
+  static final String WAIT_TIME = "zeppelin.bigquery.wait_time";
+  static final String MAX_ROWS = "zeppelin.bigquery.max_no_of_rows";
+  static final String TIME_ZONE = "zeppelin.bigquery.time_zone";
+  static final String CREDENTIAL_FILE = "zeppelin.bigquery.restricted_interpreter_credentials";
+  static final String ALLOWED_USERS = "zeppelin.bigquery.restricted_interpreter_users";
+
+  private static final char NEWLINE = '\n';
+  private static final char TAB = '\t';
+
+  private static String jobId = null;
+
+  private static final List<InterpreterCompletion> NO_COMPLETION = new ArrayList<>();
+
+  public BigQueryRestrictedAccessInterpreter(Properties property) {
+    super(property);
+  }
+
+
+  //Function to return valid BigQuery Service
+  @Override
+  public void open() {
+    if (service == null) {
+      synchronized (serviceLock) {
+        if (service == null) {
+          try {
+            service = createAuthorizedClient();
+            logger.info("Opened BigQuery SQL Connection");
+          } catch (IOException e) {
+            logger.error("Cannot open connection", e);
+            close();
+          }
+        }
+      }
+    }
+  }
+
+
+  private Boolean isAllowed(String userId) {
+
+    if (this.getProperty(ALLOWED_USERS) == null) {
+      return Boolean.TRUE;
+    }
+    logger.info(" Allowed user ids are : " + getProperty(ALLOWED_USERS));
+    if (Arrays.asList(getProperty(ALLOWED_USERS).split(",")).contains(userId)) {
```

Comment:
This is awkward.  Could just return the result of contains couldn't we?

Reply:
refactored